### PR TITLE
Sharepoint loader

### PR DESCRIPTION
**Pull request by arpitkumar980**

### CODE REVIEW

**Comment by Raj725** [comment]
@arpitkumar980 Do we need to set any env variables to run these samples?

File: pebblo_safeloader/langchain/identity-rag-sharepoint/pebblo_identity_safeload_sharepoint.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        self.folder_id = folder_id
+        self.file_id = file_id
+        self.site_id = site_id
+        self.folder_path = folder_path
+        self.qdrant_collection_name = collection_name
+
+    def load_documents(self):
+        print("\nLoading RAG documents ...")
+        loader = PebbloSafeLoader(
+            SharePointLoader(
```
Comment (Raj725) [inline pebblo_safeloader/langchain/identity-rag-sharepoint/pebblo_identity_safeload_sharepoint.py:30]: Have these samples been tested? I am getting the following error on this line:
```
pebblo/pebblo_safeloader/langchain/identity-rag-sharepoint/pebblo_identity_safeload_sharepoint.py", line 30, in load_documents
      SharePointLoader(
  File "pydantic/main.py", line 339, in pydantic.main.BaseModel.__init__
  File "pydantic/main.py", line 1066, in pydantic.main.validate_model
  File "pydantic/fields.py", line 439, in pydantic.fields.ModelField.get_default
  File "pydantic/env_settings.py", line 39, in pydantic.env_settings.BaseSettings.__init__
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 2 validation errors for _O365Settings
client_id
  field required (type=value_error.missing)
client_secret
  field required (type=value_error.missing)
```

**Review by Raj725 — COMMENTED** [review]
Please add a basic README

**Comment by arpitkumar980** [comment]
yes we need config for this

File: pebblo_saferetriever/langchain/identity-rag-sharepoint/pebblo_identity_rag_sharepoint.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        # Prepare LLM
+        self.llm = OpenAI()
+        print("Initializing PebbloRetrievalQA ...")
+        self.retrieval_chain = self.init_retrieval_chain()
+
+    def init_retrieval_chain(self):
+        """
+        Initialize PebbloRetrievalQA chain
+        """
+        return PebbloRetrievalQA.from_chain_type(
```
Comment (Raj725) [inline pebblo_saferetriever/langchain/identity-rag-sharepoint/pebblo_identity_rag_sharepoint.py:60]: @arpitkumar980 Update the initialization of `PebbloSafeLoader` and `PebbloRetrievalQA` to match the [latest sample in the main branch](https://github.com/daxa-ai/pebblo/blob/main/pebblo_saferetriever/langchain/identity-n-semantic-rag/semantic_rag.py), and then we can merge this PR.

**Review by Raj725 — CHANGES_REQUESTED** [review]
Update the samples to match the latest sample in the main branch, test it, and then we can merge this PR.

**Comment by srics** [comment]
Superseded by https://github.com/daxa-ai/pebblo/pull/399. Closing.

**Pull request closed without merging**